source/server/connection_handler_impl.h

```diff
@@ -361,7 +361,11 @@ class ActiveRawUdpListener : public Network::UdpListenerCallbacks,
   Network::Listener* listener() override { return udp_listener_.get(); }
   void pauseListening() override { udp_listener_->disable(); }
   void resumeListening() override { udp_listener_->enable(); }
-  void shutdownListener() override { udp_listener_.reset(); }
+  void shutdownListener() override {
+    // The read_filter_ should be deleted before the udp_listener_ is deleted.
```

Comment:
@danzh2010 @chadr123 sorry I haven't been tracking this PR. Can you describe in more detail what the actual problem is? I would like to understand this better, which will help build a test, but also make this comment better. My very quick understanding is that this is the line that is crashing? Is that right?

https://github.com/envoyproxy/envoy/blob/8452f4b749674deb5f4b963d0da78797fb7b1278/source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc#L247

So the issue is a race condition in which the listener is removed, but before we destroy the filter during deferred deletion. Is that right? So the fix here is to make sure that when the listener is shutdown/destroyed the filter is also destroyed?

Yeah it might not be possible to have an integration test for this case but I will determine that once we make this comment more complete in terms of what is going on here. Thank you!

/wait

Reply:
Yes. All your understanding is right.
So, the filter should be deleted before listener deletion because the filter refers to the listener.

Reply:
Ok thanks. I agree I don't see how it will be possible to make an integration test for this. Please make the comment more robust and we can ship. Thank you!

Reply:
How about this?
```suggestion
    // The read filter should be deleted before the udp listener is deleted.
    // The read filter refers to udp listener to send packets to downstream that comes from upstream.
    // So, if udp listener is deleted first than read filter, the read filter try to send packets to downstream but there is no udp listener.
    // In that case, null pointer referencing will be occurred.
    // This is serious in case of hot restart because if previous envoy process is crashed by this null pointer referencing, new envoy process that started for hot restart will be also stopped because it cannot signal to previous envoy process.
```

Reply:
Let's do this (reflow as appropriate):

```
// The read filter should be deleted before the UDP listener is deleted.
// The read filter refers to the UDP listener to send packets to downstream.
// If the UDP listener is deleted before the read filter, the read filter may try to use it after deletion.
```

Thank you!

Reply:
Ok. I have changed the comments now. :)